overlord/ifacestate/helpers.go

```diff
@@ -337,6 +337,13 @@ type connState struct {
 	DynamicPlugAttrs map[string]interface{} `json:"plug-dynamic,omitempty"`
 	StaticSlotAttrs  map[string]interface{} `json:"slot-static,omitempty"`
 	DynamicSlotAttrs map[string]interface{} `json:"slot-dynamic,omitempty"`
+	// Hotplug-related attributes: HotplugRemoved indicates a connection
+	// that disappeared because the device was removed, but may potentially
+	// be restored in the future if we see the device again.
+	// HotplugDeviceKey is the key of the associated device; it's empty for
+	// connections of regular slots.
+	HotplugRemoved   bool   `json:"hotplug-removed,omitempty"`
+	HotplugDeviceKey string `json:"hotplug-key,omitempty"`
```

Comment:
I suggest naming the first one as _HotplugGone_ instead. The difference is subtle.. "gone" feels a little more like the device was lost somehow, instead of proactively removed via snapd or similar.

The second one would be better as _HotplugKey,_ which reflects the json name  already in place, and is short and sweet.